spyder/plugins/ipythonconsole/widgets/kernelconnect.py

```diff
@@ -129,25 +131,104 @@ def __init__(self, parent=None):
         rm_layout.addWidget(auth_group)
         self.rm_group.setLayout(rm_layout)
         self.rm_group.setCheckable(True)
-        self.rm_group.setChecked(False)
         self.rm_group.toggled.connect(self.pw_radio.setChecked)
 
         # Ok and Cancel buttons
         self.accept_btns = QDialogButtonBox(
             QDialogButtonBox.Ok | QDialogButtonBox.Cancel,
             Qt.Horizontal, self)
 
+        self.accept_btns.accepted.connect(self.save_connection_settings)
         self.accept_btns.accepted.connect(self.accept)
         self.accept_btns.rejected.connect(self.reject)
 
+        # Save connection settings checkbox
+        self.save_layout = QCheckBox(self)
+        self.save_layout.setText(_("Save connection settings"))
+
+        btns_layout = QHBoxLayout()
+        btns_layout.addWidget(self.save_layout)
+        btns_layout.addWidget(self.accept_btns)
+
         # Dialog layout
         layout = QVBoxLayout(self)
         layout.addWidget(main_label)
         layout.addSpacerItem(QSpacerItem(QSpacerItem(0, 8)))
         layout.addLayout(cf_layout)
         layout.addSpacerItem(QSpacerItem(QSpacerItem(0, 12)))
         layout.addWidget(self.rm_group)
-        layout.addWidget(self.accept_btns)
+        layout.addLayout(btns_layout)
+
+        self.load_connection_settings()
+
+    def load_connection_settings(self):
+        """Load the user's previously-saved kernel connection settings."""
+        existing_kernel = CONF.get("existing-kernel", "settings", {})
+
+        connection_file_path = existing_kernel.get("json_file_path", "")
+        is_remote = existing_kernel.get("is_remote", False)
+        username = existing_kernel.get("username", "")
+        hostname = existing_kernel.get("hostname", "")
+        port = str(existing_kernel.get("port", 22))
+        is_ssh_kf = existing_kernel.get("is_ssh_keyfile", False)
+        ssh_kf = existing_kernel.get("ssh_key_file_path", "")
+
+        if connection_file_path != "":
+            self.cf.setText(connection_file_path)
+        if username != "":
+            self.un.setText(username)
+        if hostname != "":
+            self.hn.setText(hostname)
+        if ssh_kf != "":
+            self.kf.setText(ssh_kf)
+        self.rm_group.setChecked(is_remote)
+        self.pn.setText(port)
+        self.kf_radio.setChecked(is_ssh_kf)
+        self.pw_radio.setChecked(not is_ssh_kf)
+
+        try:
+            import keyring
+            ssh_passphrase = keyring.get_password("existing_kernel",
```

Comment:
Let's change this to `spyder_remote_kernel` to avoid a possible collision with another program.